save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/utils/Warning.kt

```diff
@@ -108,16 +110,22 @@ internal fun String.extractWarning(warningRegex: Regex,
     "SwallowedException")
 internal fun String.getLineNumber(warningRegex: Regex,
                                   lineGroupIdx: Int?,
+                                  placeholder: String,
+                                  lineNum: Int?,
```

Comment:
Please provide some comments about the case when `lineNum == null`, I'm already lost in all these conditions

Reply:
We pass null to the lineNum at the output when we do not need to recalculate the lines.

Reply:
Do you mean that `lineNum` is only used in placeholder mode? Then you should change `(lineNum ?: 0)` on line 125 to jsut `lineNum!!` or `requireNotNull(lineNum) { "lineNum is null" }`, because now it implies that situation with `lineNum == null` in placeholder mode is valid

Reply:
lineNum is index in _.mapIndexed{ }_ . For actualWarningsMap we send lineNum = null in getLineNumber because there we don't need to recalculate the lines

Reply:
Yes. lineNum == null in placeholder mode is invalid